code/go/0chain.net/blobbercore/handler/object_operation_handler.go

```diff
@@ -1041,24 +1038,19 @@ func (fsh *StorageHandler) WriteFile(ctx context.Context, r *http.Request) (*blo
 
 	elapsedRef := time.Since(st)
 	st = time.Now()
-	connectionObj, err := allocation.GetAllocationChanges(ctx, connectionID, allocationID, clientID)
-	if err != nil {
-		return nil, common.NewError("meta_error", "Error reading metadata for connection")
-	}
 
 	Logger.Info(fmt.Sprintf("[upload] Acquiring lock for allocation: %s, connection: %s", allocationID, connectionID))
-	mutex := lock.GetMutex(connectionObj.TableName(), connectionID)
+	mutex := lock.GetMutex(AllocationConnections, connectionID) // It will be unlocked from WithConnection();
 
 	Logger.Info(fmt.Sprintf("[upload] Locking for allocation: %s, connection: %s", allocationID, connectionID))
 
 	mutex.Lock()
 	Logger.Info(fmt.Sprintf("[upload] Acquired lock for allocation: %s, connection: %s", allocationID, connectionID))
 
-	defer func() {
```

Comment:
why is unlocking block removed?

Reply:
https://github.com/0chain/blobber/pull/1037#discussion_r1182907578